lib/tezos/README.md

```diff
@@ -0,0 +1,205 @@
+
+### Prepare to deploy nodes
+
+1. Make sure you are in the root directory of the cloned repository
+
+2. If you have deleted or don't have the default VPC, create default VPC
+
+```bash
+    aws ec2 create-default-vpc
+   ```
+
+   **NOTE:** You may see the following error if the default VPC already exists: `An error occurred (DefaultVpcAlreadyExists) when calling the CreateDefaultVpc operation: A Default VPC already exists for this account in this region.`. That means you can just continue with the following steps.
+
+   **NOTE:** The default VPC must have at least two public subnets in different Availability Zones, and public subnet must set `Auto-assign public IPv4 address` to `YES`
+
+3. Configure  your setup
+
+Create your own copy of `.env` file and edit it:
+```bash
+   # Make sure you are in aws-blockchain-node-runners/lib/tezos
+   cd lib/tezos
+   pwd
+   cp ./sample-configs/.env-sample-full.env
+   nano .env
+```
+   **NOTE:** You can find more examples inside the `sample-configs` directory.
+
+
+4. Deploy common components such as IAM role, and Amazon S3 bucket to store data snapshots
+
+```bash
+   pwd
+   # Make sure you are in aws-blockchain-node-runners/lib/tezos
+   npx cdk deploy tz-common
+```
+
+### Option 1: Single RPC Node
+
+1. Deploy Single RPC Node
+
+```bash
+   pwd
+   # Make sure you are in aws-blockchain-node-runners/lib/tezos
+   npx cdk deploy tz-single-node --json --outputs-file single-node-deploy.json
+```
+   **NOTE:** The default VPC must have at least two public subnets in different Availability Zones, and public subnet must set `Auto-assign public IPv4 address` to `YES`
+
+2. After starting the node you need to wait for the inital syncronization process to finish. It may take from an hour to half a day depending on the the state of the network. You can use Amazon CloudWatch to track the progress. To see them:
+
+    - Navigate to [CloudWatch service](https://console.aws.amazon.com/cloudwatch/) (make sure you are in the region you have specified for `AWS_REGION`)
+    - Open `Dashboards` and select `tz-single-node-<type>-<network>` from the list of dashboards.
+
+4. Once the initial synchronization is done, you should be able to access the RPC API of that node from within the same VPC. The RPC port is not exposed to the Internet. Run the following query against the private IP of the single RPC node you deployed:
+
+```bash
+   INSTANCE_ID=$(cat single-node-deploy.json | jq -r '..|.singleinstanceid? | select(. != null)')
+   NODE_INTERNAL_IP=$(aws ec2 describe-instances --instance-ids $INSTANCE_ID --query 'Reservations[*].Instances[*].PrivateIpAddress' --output text)
+
+    # We query if the node is synced to main
+    curl http://$NODE_INTERNAL_IP:8732/chains/main/is_bootstrapped
+```
+
+The result should be like this (the actual balance might change):
+
+```javascript
+   {"bootstrapped":true,"sync_state":"synced"}
+```
+
+### Option 2: Highly Available RPC Nodes
+
+1. Deploy Snapshot Node
```

Comment:
Why do we need our own snapshot node if we can download the snapshot from a public service, lie we do with single-node deployment?

Reply:
On initial sync the snapshot nodes expands the snapshot into the actual blockchain data. That takes ~40min-1h. After that, we copy the expanded files to S3. The RPC nodes can then just copy the expanded files from S3 and don't need to verify them as they come from a trusted source. Copying takes ~5min only.